packages/angularjs-workorder/lib/workorder-list/workorder-list-controller.js

```diff
@@ -19,6 +20,31 @@ function WorkorderListController($scope, workorderService, workorderFlowService,
   }
   refreshWorkorderData();
 
+
+  workorderService.on('create', function(workorder) {
+    $scope.$apply(function() {
+      self.workorders.push(workorder);
+    });
+  });
+
+  workorderService.on('remove', function(workorder) {
+    $scope.$apply(function() {
+      _.remove(self.workorders, function(w) {
+        return w.id === workorder.id;
+      });
+    });
+  });
+
+  workorderService.on('update', function(workorder) {
```

Comment:
💯  Previous solution were updating entire view using `refreshWorkorderData()`.

That this approach may mess things with filter. 
For example if you filter out workorder and then add new one - it will no longer match filter.

Reply:
For the other controler I refactored the filter to keep it updated: https://github.com/feedhenry-raincatcher/raincatcher-angularjs/pull/86/files#diff-c0c1acdaab5a4d123d6cc50a933db3daR73

Forgot to do it with this one too, nice catch!